### PR TITLE
Port jerryscript to mbedos6

### PR DESCRIPTION
**Pull request by mnegyokru**
JerryScript-DCO-1.0-Signed-off-by: Martin Negyokru mnegyokru@inf.u-szeged.hu

### CODE REVIEW

File: targets/mbedos6/Makefile [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+	rm -rf BUILD/*
+
+#clean jerry's build directory
+cleanjerry:
+	rm -rf $(JERRY_SOURCE)
+
+#install Mbed dependencies
+mbeddependencies:
+	dpkg --add-architecture i386
+	apt-get install libc6:i386 libncurses5:i386 libstdc++6:i386
```
Comment (galpeter) [inline targets/mbedos6/Makefile:109]: Do we need i386 libraries?

File: targets/mbedos6/Makefile [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+	apt install python python3 python3-pip git mercurial
+	apt install gcc-arm-none-eabi
+
+#install mbed-cli and mbed-os
+mbedinstall: mbeddependencies
+	pip3 install mbed-cli
+	python3 -m pip install -U mbed-cli
```
Comment (galpeter) [inline targets/mbedos6/Makefile:116]: This seems awkward for me: isn't the first line does the same as the second line?

File: targets/mbedos6/Makefile [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+	mbed import mbed-os
+	cd mbed-os && mbed update mbed-os-6.3.0
```
Comment (galpeter) [inline targets/mbedos6/Makefile:118]: Should there be an `mbed-os.lib` file which contains the github url which will be checked out?
AFAIK the correct command would be `mbed deploy` (if there is a mbed-os.lib file)

File: targets/mbedos6/README.md [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+Clone jerryscript:
+
+```
+git clone https://github.com/jerryscript-project/jerryscript/
+```
+Go to mbedos6 directory and run mbedconfig (you may run that as superuser), that will install dependenies and set the mbed config file:
+
+```
+cd targets/mbedos6
+sudo make mbedconfig
```
Comment (galpeter) [inline targets/mbedos6/README.md:56]: Why do we need root access for this?

File: targets/mbedos6/Makefile [hunk 5]
```diff
@@ -0,0 +1,126 @@
+##CONFIG##
+CC=GCC_ARM          #toolchain
```
Comment (galpeter) [inline targets/mbedos6/Makefile:2]: Please don't use the `CC` variable name for this.

File: targets/mbedos6/jerryscript-mbed/jerry-mbed-util/register-drivers.h [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef REGISTER_DRIVERS_H
+#define REGISTER_DRIVERS_H
+
+#include "DigitalOut-js.h"
+#include "MbedThread-js.h"
+#include "handler.h"
```
Comment (galpeter) [inline targets/mbedos6/jerryscript-mbed/jerry-mbed-util/register-drivers.h:20]: Do we need these headers at all in this file?

**Comment by rtakacs** [comment]
Closing this PR since #4909 adds support for Mbed-OS 6.15.0.

**Pull request closed without merging**